doc/manual/packages.rst

```diff
@@ -621,11 +629,22 @@ not domain experts*.
 
   * ``MATLAB.jl`` provides an interface to call the MATLAB engine from within Julia.
 
+6. Creative names are cool, but confusion isn't so much.
+
+  * Many modern programming languages such as Ruby and Python have popular packages
+    that have entertaining names. Python's ``pickle`` package isn't about pickles, it's
+    for serializing (or pickling) object structures. Ruby's ``Sinatra`` package is named
+    after the popular American singer and is a popular web framework that competes with Ruby's
+    ``Rails`` web framework (another creative name).
+
+  * If being a creative package namer isn't your style, it's okay to not force it.
```

Comment:
That's funny, I see your point. 

Creative names are for the cool kids, but if it's not your thing, it's ok. Just be descriptive and boring. (sort of paraphrasing you here).

I'm happy to remove this entire section. I thought I'd throw it in at the end of package naming guidelines to open the door to being a bit more creative. Oftentimes being descriptive devolves into an acronym party and we end up with packages like ABC.jl DEF.jl GHIK.jl and so forth. 

Given the option, I'd prefer a clever metaphor-like name to an acronym. I think they are equally less-descriptive. But when I see an acronym I cringe at having to decipher it but when I see a suggestive metaphor, I'm happy and intrigued. I _want_ to check out `pickle` and `cucumber` just because they sound cool. 

If you were to invent a javascript compiler and called it `JavaComp` it would be somewhat descriptive. And admittedly boring. But if you called it `CoffeeScript` then you will have achieved some marketing panache. 

Anyway, I'm beginning to ramble. I'm happy to keep editing away at this section and squash commits once it achieves some consensus!